src/solver/gemm_bwd.cpp

```diff
@@ -641,6 +642,13 @@ bool GemmBwdRest::IsApplicable(const ExecutionContext& context,
     if(!GemmBwdBase::IsApplicable(context, problem))
         return false;
 
+#if WORKAROUND_ISSUE_1956
+    // Col2Im3dGPU has issues with address computations when workspace size > 4GiB.
+    if(problem.GetConv().GetSpatialDimension() == 3 &&
+       GetWorkspaceSize(context, problem) > 0xffffffffULL)
+        return false;
+#endif
```

Comment:
ℹ️ This is the W/A. Other changes relate to by-products (logging improvements etc).

Reply:
Yep and we need to remove this WA when fully implemented

Reply:
@junliume Of course and that is the reason why we use tagging like WORKAROUND_ISSUE_1956 ))) Thanks for reviewing!

Reply:
@junliume BTW theoretically the reason could still be a bug in the OCL compiler. If this is the case, then we'll make the value of WORKAROUND_ISSUE_1956 dependent on ROCm version.